packages/bsky/src/api/app/bsky/feed/getPostThread.ts

```diff
@@ -106,11 +111,20 @@ const presentation = (
     height: params.parentHeight,
     depth: params.depth,
   })
+
+  let rootUri: string | undefined = undefined
+  if (isThreadViewPost(thread) && isPostRecord(thread.post.record)) {
```

Comment:
it may be a bit simpler to just get the rootUri off of the hydration state with the help of params rather than off the thread view. Something like:
`const rootUri = hydration.posts?.get(params.uri)?.record.reply?.root ?? params.uri`

Reply:
Ah yeah, that was kinda my initial thought too, but `params.uri` could be a child within a thread too, and we then return its parents. So basically need to find any post within a thread and get its reply root. And if it's the root post in the thread, then there's no `ReplyRef`.

Reply:
hmmm I think there's 4 cases:
- `params.uri` is the root & the record is found, in which case the record won't have a reply root set and we'll end up using `params.uri`
- `params.uri` is the root & the record is not found, in which case we'll just end up using the `params.uri`
- `params.uri` is a child & the record is found, in which case, it should have reply root set (and the dataplane will hinge it's thread construction off that reply root) so we'll end up using reply root
- `params.uri` is a child & the record is not found, in which case we'll end up using `params.uri`

I believe the first 3 are correct. But you're right that the 4th is incorrect. Maybe we could check if the post is found and if not then rootUri is `undefined`? That's essentially the same logic you have just driving it off of the hydration map instead of off the view.

Reply:
Oy, yes, misunderstood this. This is much cleaner!